app/rdf/query-dimension-values.ts

```diff
@@ -232,39 +232,59 @@ const getQueryFilters = (
   cube: ExtendedCube,
   dimensionIri: string | undefined
 ) => {
-  return filtersList.length > 0
-    ? filtersList.map(([iri, value], i) => {
-        const dimension = cube.dimensions.find((d) => d.path?.value === iri);
-
-        // Ignore the current dimension
-        if (!dimension || dimensionIri === iri) {
-          return "";
-        }
-
-        // Ignore filters with no value or with the special value
-        if (
-          value.type === "single" &&
-          (value.value === FIELD_VALUE_NONE || isDynamicMaxValue(value.value))
-        ) {
-          return "";
-        }
-
-        // Ignore range filters for now
-        if (value.type === "range") {
-          return "";
-        }
-
-        const versioned = dimension ? dimensionIsVersioned(dimension) : false;
-
-        return sparql`${
-          versioned
-            ? sparql`?dimension${i} ${ns.schema.sameAs} ?dimension_unversioned${i}.`
-            : ""
-        }
-      ?observation <${iri}> ?dimension${i}.
-      ${formatFilterIntoSparqlFilter(value, dimension, versioned, i)}`;
-      })
-    : "";
+  if (filtersList.length === 0) {
+    return "";
+  }
+
+  let i = 0;
+  const filterDimensionIris = filtersList.map(([iri]) => iri);
+  // Also include other dimensions to make sure we don't return values that could
+  // result in no observations
```

Comment:
I fail to see the previous buggy case ? Could you explain to me why if we do not add those dimensions, there is a bug 😅  ?

Reply:
Sure, take a look at these queries:
- before change (7 results): https://s.zazuko.com/4ntKoP,
- after change (5 results): https://s.zazuko.com/xohXms.

Also check the number 5 in `How to test (TEST)` section from the PR description 👍 

Basically if we don't include other dimensions, we risk that there is an observation that contains values for the "core dimensions", but has some undefined values in other dimensions – which leads to a "reload" of possible filters and the bug in the PR description. Inclusion of other dimensions act as a filter to remove undefined values potentially attached to these other dimensions.

Let me know if I can clarify this further 👀